core/src/main/java/org/apache/calcite/runtime/SqlFunctions.java

```diff
@@ -2662,6 +2662,37 @@ public static void jsonArrayAggAdd(List list, Object element,
     }
   }
 
+  public static String jsonType(Object input) {
+    String result = "";
```

Comment:
Would it be better to make variable `result` final and assign values in if branches? So it need not to be assigned twice.